### ISSUE TITLE
Cardano RPC, isEnabled never resolves for non connected wallets?

### ISSUE THREAD

**Issue by ObjectAssign** [cardano-foundation/developer-portal#1235]
Hello, i'm implementing Cardano wallet login on Bitculator.

The problem is that i want to check if a wallet is already connected (not necessarily signed)... i see a method called isEnabled in the injected window.cardano[walletname] object which is a js promise method.

The problem is, it only resolves for wallets that are already connected, it never resolves for wallets not connected?

I'm still a newbie on this subject, but i'm wondering if it is a general problem or if it is poorly implemented by the wallet devs?

Here is the snippet:

```
async function setWalletProviders()
{
	const providers = [];
	for (const key in window.cardano) {
		if (key !== 'enable' && key !== 'isEnabled') {
			console.log('WINDOW CARDANO: ', window.cardano[key]);

			providers.push({
				name: key,
				isConnected: window.cardano[key].isEnabled(),
				...window.cardano[key]
			});
		}
	}

	console.log('IAM providers: ', providers);
	walletProviders.value.push(...providers);
}
```

Also is there a better way to detect the wallets than excluding all keys "enable" and "isEnable" from the root window.cardano object? this approach seems a bit fluffy because what if a new key is added to the object, is there really no standarized model for this? seems pretty simple to me to implement a standard that lets wallets inject their object into a "wallets" key in the window.cardano object, or something similar....

**Comment by ObjectAssign** [comment]
I made a helper method that wraps the promise like this:

export function resolveWithTimeout(promise, timeoutMs = 1000) {
	let timeoutHandle;
	const timeoutPromise = new Promise((resolve, reject) => {
		timeoutHandle = setTimeout(() => reject(new Error('Promise timed out')), timeoutMs);
	});

	return Promise.race([promise, timeoutPromise])
		.then(result => {
			clearTimeout(timeoutHandle);
			return result;
		})
		.catch(error => {
			clearTimeout(timeoutHandle);
			throw error;
		});
}

However, i discovered that it "hangs" and never resolves when you have a wallet currently awaiting your approval to connect.

So basically, there is no problem and this thread can be closed :) (still interested in a better way to discover all wallets tho)
